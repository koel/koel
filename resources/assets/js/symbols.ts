import { DeepReadonly, InjectionKey, Ref } from 'vue'
import Overlay from '@/components/ui/Overlay.vue'
import DialogBox from '@/components/ui/DialogBox.vue'
import MessageToaster from '@/components/ui/MessageToaster.vue'
import Router from '@/router'

export type ReadonlyInjectionKey<T> = InjectionKey<[Readonly<T> | DeepReadonly<T>, Closure]>

export const RouterKey: InjectionKey<Router> = Symbol('Router')
export const OverlayKey: InjectionKey<Ref<InstanceType<typeof Overlay>>> = Symbol('Overlay')
export const DialogBoxKey: InjectionKey<Ref<InstanceType<typeof DialogBox>>> = Symbol('DialogBox')
export const MessageToasterKey: InjectionKey<Ref<InstanceType<typeof MessageToaster>>> = Symbol('MessageToaster')

export const SongsKey: ReadonlyInjectionKey<Ref<Song[]>> | InjectionKey<Ref<Song[]>> = Symbol('Songs')
export const CurrentSongKey: InjectionKey<Ref<Song | undefined>> = Symbol('CurrentSong')
export const SelectedSongsKey: ReadonlyInjectionKey<Ref<Song[]>> = Symbol('SelectedSongs')
export const SongListConfigKey: ReadonlyInjectionKey<Partial<SongListConfig>> = Symbol('SongListConfig')
export const SongListSortFieldKey: ReadonlyInjectionKey<Ref<SongListSortField>> = Symbol('SongListSortField')
export const SongListSortOrderKey: ReadonlyInjectionKey<Ref<SortOrder>> = Symbol('SongListSortOrder')
export const SongListFilterKeywordsKey: InjectionKey<Ref<string>> = Symbol('SongListFilterKeywords')

export const ModalContextKey: InjectionKey<Ref<Record<string, any>>> = Symbol('ModalContext')
