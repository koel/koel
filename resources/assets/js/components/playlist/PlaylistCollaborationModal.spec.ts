import { ref } from 'vue'
import { expect, it } from 'vitest'
import factory from '@/__tests__/factory'
import UnitTestCase from '@/__tests__/UnitTestCase'
import { ModalContextKey } from '@/symbols'
import Modal from './PlaylistCollaborationModal.vue'

new class extends UnitTestCase {
  protected test () {
    it('renders the modal', async () => {
      const { html } = this.render(Modal, {
        global: {
          provide: {
            [<symbol>ModalContextKey]: [ref({ playlist: factory('playlist') })],
          },
          stubs: {
            InviteCollaborators: this.stub('InviteCollaborators'),
            CollaboratorList: this.stub('CollaboratorList'),
          },
        },
      })

      expect(html()).toMatchSnapshot()
    })
  }
}
