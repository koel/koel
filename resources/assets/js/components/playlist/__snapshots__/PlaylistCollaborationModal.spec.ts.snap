// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`playlistCollaborationModal.vue > renders the modal 1`] = `
<div data-v-886145d2="" class="collaboration-modal max-w-[640px]" tabindex="0" data-testid="playlist-collaboration">
  <header data-v-886145d2="">
    <h1 data-v-886145d2="">Playlist Collaboration</h1>
  </header>
  <main data-v-886145d2="">
    <p data-v-886145d2="" class="text-k-text-secondary"> Collaborative playlists allow multiple users to contribute. <br data-v-886145d2=""> Note: Songs added to a collaborative playlist are made accessible to all users, and you cannot mark a song as private if itâ€™s still part of a collaborative playlist. </p>
    <section data-v-886145d2="" class="space-y-5">
      <h2 data-v-886145d2="" class="flex text-xl mt-6 mb-1 items-center"><span data-v-886145d2="" class="flex-1">Current Collaborators</span>
        <!--v-if-->
      </h2>
      <div data-v-886145d2="" class="collaborators-wrapper overflow-auto"><br data-v-886145d2="" data-testid="CollaboratorList" playlist="[object Object]"></div>
    </section>
  </main>
  <footer data-v-886145d2=""><button data-v-8943c846="" data-v-886145d2="" class="text-base text-k-text-primary bg-k-primary px-3.5 py-2 rounded cursor-pointer" type="button">Close</button></footer>
</div>
`;
